src/main/scala/com/databricks/spark/csv/CsvRelation.scala

```diff
@@ -120,12 +120,26 @@ case class CsvRelation protected[spark] (
       schemaFields: Seq[StructField],
       projection: MutableProjection,
       row: GenericMutableRow): Iterator[Row] = {
-    iter.map { line =>
-      val tokens = CSVParser.parse(line, csvFormat).getRecords.head
-      for (index <- 0 until schemaFields.length) {
-        row(index) = tokens.get(index)
+    iter.flatMap { line =>
+      try {
+        val records = CSVParser.parse(line, csvFormat).getRecords
+        if (records.isEmpty) {
+          logger.warn(s"Ignoring empty line: $line")
+          None
+        } else {
+          val tokens = records.head
+          var index = 0
+          while (index < schemaFields.length) {
+            row(index) = tokens.get(index)
+            index = index + 1
+          }
+          Some(projection(row))
+        }
+      } catch {
+        case e: Exception =>
```

Comment:
you probably want to match NonFatal http://www.scala-lang.org/files/archive/nightly/docs/library/index.html#scala.util.control.NonFatal$